tests/test_tuya_builder.py

```diff
@@ -74,6 +77,65 @@ async def test_convenience_methods(device_mock, method_name, attr_name, exp_clas
     assert isinstance(ep_attr, exp_class)
 
 
+@pytest.mark.parametrize(
+    "power_cfg,battery_type,battery_qty,battery_voltage",
+    [
+        (TuyaPowerConfigurationCluster2AAA, None, None, None),
+        (None, BatterySize.CR123A, 1, 60),
+        (None, BatterySize.CR123A, 1, None),
+        (None, BatterySize.AA, None, None),
+        (None, None, None, None),
+    ],
+)
+async def test_battery_methods(
+    device_mock, power_cfg, battery_type, battery_qty, battery_voltage
+):
+    """Test the battery convenience method."""
+
+    registry = DeviceRegistry()
+
+    (
+        TuyaQuirkBuilder(device_mock.manufacturer, device_mock.model, registry=registry)
+        .tuya_battery(
+            dp_id=1,
+            power_cfg=power_cfg,
+            battery_type=battery_type,
+            battery_qty=battery_qty,
+            battery_voltage=battery_voltage,
+        )
+        .tuya_onoff(dp_id=3)
+        .skip_configuration()
+        .add_to_registry()
+    )
+
+    quirked = registry.get_device(device_mock)
+    ep = quirked.endpoints[1]
+
+    assert ep.power is not None
+
+    if power_cfg:
+        assert isinstance(ep.power, TuyaPowerConfigurationCluster2AAA)
+        assert ep.power.get("battery_size") == BatterySize.AAA
+        assert ep.power.get("battery_quantity") == 2
+        assert ep.power.get("battery_rated_voltage") == 15
+
+    elif battery_type or battery_qty or battery_voltage:
+        if battery_type:
+            assert ep.power.get("battery_size") == battery_type
+        else:
+            assert ep.power.get("battery_size") == BatterySize.AA
+
+        if battery_qty:
+            assert ep.power.get("battery_quantity") == battery_qty
+        else:
+            assert ep.power.get("battery_quantity") is None
+
+        if battery_voltage:
+            assert ep.power.get("battery_rated_voltage") == battery_voltage
+        elif battery_qty:
+            assert ep.power.get("battery_rated_voltage") == 30
```

Comment:
It might be cleaner to put the expected values into `parametrize` as well.
I'll do that quickly, since I already have it open anyway now.

Reply:
Done with https://github.com/zigpy/zha-device-handlers/pull/3728/commits/32086b58847223e81b926ff5428dba9df0286e50

Reply:
We're only checking for instance of `TuyaPowerConfigurationCluster` (always now), not specifically for `TuyaPowerConfigurationCluster2AAA` to avoid an if branch in the test. Should be fine, since we already test the values for that.

Reply:
Ok, I thought `TuyaPowerConfigurationCluster2AAA ` was alsos of `TuyaPowerConfigurationCluster `, but it's obviously not. Fixed that with https://github.com/zigpy/zha-device-handlers/pull/3728/commits/6f9574373781122748980f8f11640738580866b7 and we're now checking for the correct cluster again.

Reply:
Cleaned up slightly more with https://github.com/zigpy/zha-device-handlers/pull/3728/commits/039a47991aed641036ba49d1618d664ed90f5424